test/node/bson_corpus.spec.test.js

```diff
@@ -181,22 +186,29 @@ describe('BSON Corpus', function () {
               // convert inputs to native Javascript objects
               const nativeFromCB = bsonToNative(cB);
 
-              if (cEJ.includes('1.2345678921232E+18')) {
+              if (description === 'Double type') {
                 // The following is special test logic for a "Double type" bson corpus test that uses a different
                 // string format for the resulting double value
                 // The test does not have a loss in precision, just different exponential output
                 // We want to ensure that the stringified value when interpreted as a double is equal
                 // as opposed to the string being precisely the same
-                if (description !== 'Double type') {
-                  throw new Error('Unexpected test using 1.2345678921232E+18');
-                }
                 const eJSONParsedAsJSON = JSON.parse(cEJ);
                 const eJSONParsed = EJSON.parse(cEJ, { relaxed: false });
                 expect(eJSONParsedAsJSON).to.have.nested.property('d.$numberDouble');
                 expect(eJSONParsed).to.have.nested.property('d._bsontype', 'Double');
                 const testInputAsFloat = Number.parseFloat(eJSONParsedAsJSON.d.$numberDouble);
+                const testInputAsNumber = Number(eJSONParsedAsJSON.d.$numberDouble);
                 const ejsonOutputAsFloat = eJSONParsed.d.valueOf();
-                expect(ejsonOutputAsFloat).to.equal(testInputAsFloat);
+                if (eJSONParsedAsJSON.d.$numberDouble === 'NaN') {
+                  expect(ejsonOutputAsFloat).to.be.NaN;
+                  expect(ejsonOutputAsFloat).to.be.NaN;
```

Comment:
duplicate assertion?
```suggestion
                  expect(ejsonOutputAsFloat).to.be.NaN;
```

Reply:
Yep, fixed